moveos/moveos-verifier/src/metadata.rs

```diff
@@ -270,6 +275,85 @@ impl<'a> ExtendedChecker<'a> {
 // Entry Function
 
 impl<'a> ExtendedChecker<'a> {
+    fn check_init_module(&mut self, module: &ModuleEnv) {
+        for ref fun in module.get_functions() {
+            if fun.get_identifier().as_ident_str() != INIT_FN_NAME_IDENTIFIER.as_ident_str() {
+                continue;
+            }
+
+            if Visibility::Private != fun.visibility() {
+                self.env
+                    .error(&fun.get_loc(), "module init function should private")
+            }
+
+            if fun.is_entry() {
+                self.env
+                    .error(&fun.get_loc(), "module init function should not entry")
+            }
+
+            if fun.get_return_count() != 0 {
+                self.env.error(
+                    &fun.get_loc(),
+                    "module init function should not have return",
+                )
+            }
+
+            let arg_tys = &fun.get_parameter_types();
+            if arg_tys.len() != 1 && arg_tys.len() != 2 {
+                self.env.error(
+                    &fun.get_loc(),
+                    "module init function only should have a parameter",
+                )
+            }
+            for ty in arg_tys {
+                match ty {
+                    
+                    Type::Reference(true, bt) => {
+                        
+                        let struct_tag = bt.clone().into_struct_tag(self.env);
+                        if struct_tag.is_none() {
+                            self.env.error(
+                                &fun.get_loc(),
+                                "module init function only should have two parameter with signer or storageContext"
```

Comment:
Similar multiple error messages like the following should be put at the same exit instead of being repeated multiple times.

`"module init function only should have two parameter with signer or storageContext"`

Reply:
Good suggestion
It has been changed to a different error message
Which will help developers have a clearer understanding of their code